app/Models/ProviderUserMapping.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class ProviderUserMapping extends Model
+{
+    use HasFactory;
+
+    protected $fillable = [
+        'user_id',
+        'provider_platform_id',
+        'external_user_id',
+        'external_username', // if the user has a field in the external platform that is different from 'username'
+        'authentication_provider_id',
```

Comment:
This might need to go on the provider platform or might not be needed at all.

Reply:
Yeah this will depend on whether it is possible that a 'login' for a user from our platform would ever have a login assigned to anything but our oidc server. If this exists on the Provider Platform model, then I think it would need to be nullable. 

Hard to tell at this point what unique info we may need about a user related to a provider login elsewhere